psh/df/df.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Phoenix-RTOS
+ *
+ * df - print filesystem statistics
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <sys/statvfs.h>
+
+#include "../psh.h"
+
+
+static void psh_df_info(void)
+{
+	printf("print filesystem statistics");
+}
+
+
+static void psh_df_help(const char *prog)
+{
+	printf("Usage: %s [options] [files]\n", prog);
+	printf("  -T:  print filesystem type\n");
+	printf("  -i:  print inode information instead of block usage\n");
+	printf("  -h:  print this help message\n");
+}
+
+
+static int psh_df(int argc, char **argv)
+{
+	int i, c, type = 0, inodes = 0;
+	struct statvfs st;
+	fsblkcnt_t used;
+
+	while ((c = getopt(argc, argv, "Tih")) != -1) {
+		switch (c) {
+			case 'T':
+				type = 1;
+				break;
+
+			case 'i':
+				inodes = 1;
+				break;
+
+			case 'h':
+				psh_df_help(argv[0]);
+				return EXIT_SUCCESS;
+
+			default:
+				psh_df_help(argv[0]);
+				return EXIT_FAILURE;
+		}
+	}
+
+	if (optind == argc) {
+		printf("df: no mount table support\n");
+		return EXIT_FAILURE;
+	}
+
+	printf("Filesystem     %s %-9s      Used Available Capacity Mounted on\n",
+		(type) ? " Type      " : "",
+		(inodes) ? "   Inodes" : "1K-blocks");
+
+	for (i = optind; i < argc; ++i) {
+		if (statvfs(argv[i], &st) < 0) {
+			fprintf(stderr, "df: %s: no such file or directory\n", argv[i]);
+			continue;
```

Comment:
Imho exit code should be EXIT_FAILURE if there's at least one fail

Reply:
Fixed (I verified that behaviour on Linux and returned exit code is indeed non-zero).

Reply:
No need for errno, we ignore it's value anyway by returning either EXIT_FAILURE or EXIT_SUCCESS